x2gbfs/providers/cambio.py

```diff
@@ -0,0 +1,153 @@
+                '_city': address.get('addressLocation'),  # Non-standard
+                'rental_uris': rental_uris,
+            }
+
+            gbfs_station_status = self._create_station_status(station_id, default_last_reported)
+            gbfs_station_status.update(
+                {
+                    'vehicle_types_available': vehicle_types_available,
+                    'num_bikes_available': len(vehicle_types_available),
+                }
+            )
+
+            gbfs_station_infos_map[station_id] = gbfs_station
+            gbfs_station_status_map[station_id] = gbfs_station_status
+
+        return gbfs_station_infos_map, gbfs_station_status_map
+
+    def load_vehicles(self, default_last_reported: int) -> Tuple[Optional[Dict], Optional[Dict]]:
+        """
+        Returns vehicle_types_map (with a single vehicle type)
+        and vehicles_map, both keyed by the vehicle's/vehicle type's ID.
+        It's values must be dicts conforming to GBFS Spec v2.3 vehicles / vehicle types.
+        For details, see https://github.com/MobilityData/gbfs/blob/v2.3/gbfs.md
+        """
+
+        result = self._all_vehicle_types()
+
+        gbfs_vehicle_types_map: dict[str, dict] = {}
+        gbfs_vehicles_map: dict[str, dict] = {}
+        for elem in result:
+            # Note: usually you would iterate over a data source retrieve from a proprietary
+            # system and convert it to vehicles, reflectig theit real-time properties.
+            vehicle_type_id = elem['id']
+            propulsion_type = self._extract_propulsion_type(elem)
+
+            gbfs_vehicle_type = {
+                'vehicle_type_id': vehicle_type_id,
+                'name': elem['displayName'],
+                'form_factor': 'car',
+                'default_pricing_plan': elem['priceClass']['id'],
+                'propulsion_type': propulsion_type,
+                'max_range_meters': (
+                    self.DEFAULT_MAX_RANGE_ELECTRIC
+                    if propulsion_type == 'electric'
+                    else self.DEFAULT_MAX_RANGE_COMBUSTION
+                ),
+                'wheel_count': 4,
+                'return_constraint': 'roundtrip_station',
+            }
+            gbfs_vehicle_types_map[vehicle_type_id] = gbfs_vehicle_type
+
+        return gbfs_vehicle_types_map, gbfs_vehicles_map
+
+    def _extract_vehicle_types_available(self, elem: dict) -> list[dict]:
+        """
+        Extracts a list of vehicle_types_available from station. As it is static,
+        we declare an availability of 1 per vehicle_type, though this incorrect.
+        """
+        vehicle_classes_at_station = elem.get('vehicleClasses', [])
+        return [{'vehicle_type_id': vehicle_class['id'], 'count': 1} for vehicle_class in vehicle_classes_at_station]
+
+    def _replace_umlauts(self, string: str) -> str:
+        substitions = (('ü', 'ue'), ('ä', 'ae'), ('ö', 'oe'), ('é', 'e'), ('ß', 'ss'))
```

Comment:
Can you be sure that there are just these Umlauts? Maybe at least make sure with an regexp to remove all other possible not-allowed characters?

Reply:
Aren't there PyPI packages for this?

Reply:
60ad0a79e9de0a70427d21cfa7aa426e9945c1bc switches to PyPI package [unidecode](https://pypi.org/project/Unidecode/). As it explicitly not respects German umlaut replacement rules and never will, I added a utility function which performs German replacements before passing the string to unidecode (as suggested in unidecode's FAQ)